### PR TITLE
protoc-gen-openapiv2: Remove path parameters from body when body is a snake_case field

### PR DESCRIPTION
**Pull request by oyvindwe**
#### References to other Issues or PRs

Fixes a case missed in PR #2553

#### Have you read the [Contributing Guidelines](https://github.com/grpc-ecosystem/grpc-gateway/blob/master/CONTRIBUTING.md)?

Yes.

#### Brief description of what is fixed or changed

If the JSON representation of the body field name is different from the native proto field name, path parameters were not removed from the body.

#### Other comments

None.

### CODE REVIEW

File: protoc-gen-openapiv2/internal/genopenapi/template.go [hunk 1]
```diff
@@ -1164,7 +1164,7 @@ func renderServices(services []*descriptor.Service, paths openapiPathsObject, re
 							bodyFieldName = bodyField.Name
 						}
 						// Align pathParams with body field path.
-						pathParams := subPathParams(bodyFieldName, b.PathParams)
+						pathParams := subPathParams(bodyField.Name, b.PathParams)
```
Comment (johanbrandhorst) [inline protoc-gen-openapiv2/internal/genopenapi/template.go:1167]: Are there no test changes required for this? Do we have any test cases that cover this?

Comment (oyvindwe) [reply protoc-gen-openapiv2/internal/genopenapi/template.go:1167]: > Are there no test changes required for this? Do we have any test cases that cover this?

I will add a test case.

Comment (oyvindwe) [reply protoc-gen-openapiv2/internal/genopenapi/template.go:1167]: @johanbrandhorst test case added!

**Review by johanbrandhorst — APPROVED** [review]

**Comment by johanbrandhorst** [comment]
Thanks for your contribution! I think this warrants a patch release!

**Comment by oyvindwe** [comment]
> Thanks for your contribution! I think this warrants a patch release!

I'm happy to help out! Unfortunately this is a side project, so it took a while to get the unit test in place, but I should have made that as a part of PR #2553.

A patch release would be great!

**Pull request merged**